src/Flux/neural_ode.jl

```diff
@@ -0,0 +1,20 @@
+neural_ode_reduction(sol) = Array(sol)
+neural_ode(x,model,tspan,args...;kwargs...) = neural_ode(x,model,tspan,
+                                                         diffeq_adjoint,
+                                                         args...;kwargs...)
+function neural_ode(x,model,tspan,
+                    ad_func::Function,
+                    args...;kwargs...)
+  p = Flux.data(destructure(model))
+  dudt_(du,u::TrackedArray,p,t) = du .= restructure(model,p)(u)
+  dudt_(du,u::AbstractArray,p,t) = du .= Flux.data(restructure(model,p)(u))
+  prob = ODEProblem(dudt_,x,tspan,p)
+
+  if ad_func === diffeq_adjoint

```

Comment:
This feels like it should be handled with a dispatch
more extensible that way.

Won't these methods fit onto `diffeq_adjoint` and `diffeq_fd` ?

Even if not 
a suitable proxy can be betteren.

`neural_ode_ad_func(::typeof(f), p, prob, args..., kwargs...)`

Reply:
I'll merge for now, but a follow-up PR changing the API around is fine. I just want to get the internals solidified.